elastic/kibana
# [Response Ops][Event Log] Updating event log mappings if data stream and index template already exist

Resolves https://github.com/elastic/kibana/issues/192682

## Summary

As of 8.8, we started writing all event log documents to the `.kibana-event-log-ds` index. Prior to this, we created a new index template and data stream for every version (`.kibana-event-log-8.7` for example) so any mapping updates that were added for the version were created in the new index on upgrade.

With the static index name and serverless, we need a way to update mappings in existing indices. This PR uses the same mechanism that we use for the alerts index to update the index template mappings and the mappings for the concrete backing indices of a datastream.

## To Verify

Run ES and Kibana in `main` to test the upgrade path for serverless 
  a. Check out `main`, run ES: `yarn es snapshot --license trial --ssl -E path.data=../test_el_upgrade` and Kibana `yarn start --ssl` 
  b. Create a rule and let it run to populate the event log index
  c. Switch to this PR branch. Make a mapping update to the event log index:

```
--- a/x-pack/plugins/event_log/generated/mappings.json
+++ b/x-pack/plugins/event_log/generated/mappings.json
@@ -172,6 +172,9 @@
         },
         "rule": {
             "properties": {
+                "test": {
+                  "type": "keyword"
+                },
                 "author": {
                     "ignore_above": 1024,
                     "type": "keyword",
```
  d. Start ES and Kibana with the same commands as above
  e. Verify that the `.kibana-event-log-ds` index is created and has the updated mapping: 
     - https://localhost:5601/app/management/data/index_management/templates/.kibana-event-log-template
     - https://localhost:5601/app/management/data/index_management/indices/index_details?indexName=.ds-.kibana-event-log-ds-2024.09.17-000001&filter=.kibana-&includeHiddenIndices=true&tab=mappings

I also verified the following:
1. Run ES and Kibana in 8.7 to test the upgrade path from 8.7 (when event log indices were versioned) to now
2. Run ES and Kibana in 8.15 to test the upgrade path from the previous release to now

However, I had to create an 8.x branch and cherry pick this commit because `main` is now on 9.0 and we can't upgrade directly from older 8.x version to 9.0!
